src/main/java/com/rtb/projectmanagementtool/project/ProjectData.java

```diff
@@ -0,0 +1,204 @@
+   */
+  public ProjectData(String name, String description, long creatorId) {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    this.id = datastore.put(new Entity("Project")).getId();
+    this.name = name;
+    this.description = description;
+
+    this.userIds = new HashMap<Long, User>();
+    this.userIds.put(creatorId, User.CREATOR);
+
+    this.taskIds = new HashSet<Long>();
+    toEntity();
+  }
+
+  /**
+   * Class constructor to create Project from existing datastore entity
+   *
+   * @param entity the datastore entity to parse object from
+   */
+  public ProjectData(Entity entity) {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    this.id = (Long) entity.getKey().getId();
+    this.name = (String) entity.getProperty(PROPERTY_NAME);
+    this.description = (String) entity.getProperty(PROPERTY_DESCRIPTION);
+
+    Gson gson = new Gson();
+
+    // sets the type of containers to store parsed JSON in
+    Type mapType = new TypeToken<HashMap<Long, User>>() {}.getType();
+    Type setType = new TypeToken<HashSet<Long>>() {}.getType();
+
+    this.userIds = gson.fromJson((String) entity.getProperty(PROPERTY_USER_IDS), mapType);
+    this.taskIds = gson.fromJson((String) entity.getProperty(PROPERTY_TASK_IDS), setType);
+  }
+
+  /** Saves the object as a datastore entity. */
+  private void toEntity() {
+    Gson gson = new Gson();
+    try {
+      Key entityKey = KeyFactory.createKey("Project", this.id);
+      Entity entity = datastore.get(entityKey);
+
+      entity.setProperty(PROPERTY_NAME, this.name);
+      entity.setProperty(PROPERTY_DESCRIPTION, this.description);
+      entity.setProperty(PROPERTY_USER_IDS, gson.toJson(this.userIds));
+      entity.setProperty(PROPERTY_TASK_IDS, gson.toJson(this.taskIds));
+
+      datastore.put(entity);
+    } catch (Exception exception) {
+      System.out.println("Error: Project not found.");
+    }
+  }
+
+  /** @return project id */
+  public long getId() {
+    return this.id;
+  }
+
+  /** @return project name */
+  public String getName() {
+    return this.name;
+  }
+
+  /** @return project description */
+  public String getDescription() {
+    return this.description;
+  }
+
+  /** @return returns user ids */
+  public HashMap<Long, User> getUsers() {
+    return this.userIds;
+  }
+
+  /** @return task ids */
+  public HashSet<Long> getTaskIds() {
+    return this.taskIds;
+  }
+
+  /**
+   * Changes the name of this project
+   *
+   * @param name the new name of the project
+   */
+  public void setName(String name) {
+    this.name = name;
+    toEntity();
+  }
+
+  /**
+   * Changes the description of this project
+   *
+   * @param description the new description of the project
+   */
+  public void setDescription(String description) {
+    this.description = description;
+    toEntity();
+  }
+
+  /**
+   * Adds a user to the project ? separate this into addAdminUser and addRegularUser?
```

Comment:
I'm in favor of separating into two methods. Passing booleans around quickly gets tedious and sometime dangerous (i.e. if a caller accidentally passes two boolean values into a method out of order it can lead to unintended behavior).